app/page/postRank.js

```diff
@@ -0,0 +1,184 @@
+const nest = require('depnest')
+const { h, Value, computed, when } = require('mutant')
+const Scroller = require('pull-scroll')
+const next = require('pull-next-query')
+const pull = require('pull-stream')
+
+exports.gives = nest({
+  'app.html.menuItem': true,
+  'app.page.postRank': true
+})
+
+exports.needs = nest({
+  'app.html.scroller': 'first',
+  'app.sync.goTo': 'first',
+  'keys.sync.id': 'first',
+  'channel.obs.subscribed': 'first',
+  'contact.obs.following': 'first',
+  'sbot.pull.log': 'first',
+  'message.html.render': 'first'
+})
+
+exports.create = function (api) {
+  return nest({
+    'app.html.menuItem': menuItem,
+    'app.page.postRank': page
+  })
+
+  function menuItem () {
+    return h('a', {
+      style: { order: 1 },
+      'ev-click': () => api.app.sync.goTo({ page: 'postRank' })
+    }, '/postRank')
+  }
+
+  function getPosts(channelPostBoost, friendPostBoost, friendVoteBoost, friendCommentBoost, threshold, cb)
+  {
+    let weights = {
+      post: 1, // base
+      channelPost: channelPostBoost,
+      friendPost: friendPostBoost,
+
+      vote: 0.5, // base
+      friendVote: friendVoteBoost,
+
+      comment: 1, // base
+      friendComment: friendCommentBoost
+    }
+
+    console.log("weights", weights)
+
+    const myKey = api.keys.sync.id()
+    let myChannels = api.channel.obs.subscribed(myKey)()
+    let imFollowing = new Set(api.contact.obs.following(myKey)())
+
+    console.log(myChannels)
+    console.log(imFollowing)
+
+    var messages = {}
+    
+    console.log("pulling", new Date())
+
+    return pull(
+      api.sbot.pull.log({ reverse: true, limit: 10000 }),
```

Comment:
This is fine for a first pass.


I would _highly_ recommend using ssb-query, and fetching just the `post`  and `vote` messages and merging with `pull-merge`.

Can also add a next-stepper I built for ssb-query called `pull-next-query` which means it will only load 100 messages at a time rather than a full 10000 into the UI.

If you want any help setting that stuff up let me know

Reply:
Thanks!

This was mostly just porting over some existing code I wrote a while ago for fun. I've added the missing mcss. Forgot to check that in.